### PR TITLE
MGMT-9528: remove dependency on docker invocation for code generation

### PR DESCRIPTION
**Pull request by osherdp**
This change aims to refactor how we generate code. Instead of spawning docker containers for each of the tasks (generating go client, generating Python client, generating backend-related files, etc.) it will incorporate relevant tools/binaries from external images into the ``assisted-service-build`` image so that it can be used in a prow job (that cannot spawn docker-in-docker containers).

With those changes, we should be able to introduce a job named ``verify-generated-code`` which makes sure that the developer of a PR has generated all relevant code responding with the PR changes.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [x] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @eliorerz @adriengentil @filanov 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: Dockerfile.assisted-service-build [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
 ENV PATH=$PATH:$GOROOT/bin:$GOPATH/bin
 
 USER 0
 
+RUN mkdir build && chmod g+xw -R build/
+
 RUN yum install -y epel-release \
     && yum install -y --setopt=skip_missing_names_on_install=False \
-        gcc genisoimage jq git libvirt-client libvirt-devel
+        gcc genisoimage jq git libvirt-client libvirt-devel java
```
Comment (osherdp) [inline Dockerfile.assisted-service-build:16]: father please forgive my sins

**Comment by adriengentil** [comment]
Is this new job in Prow will call `make ci-lint` like in Jenkins? In that case, we may want to remove the call to skipper in the target.

File: Dockerfile.assisted-service-build [hunk 2]
```diff
@@ -1,19 +1,23 @@
 FROM quay.io/edge-infrastructure/postgresql-12-centos7:latest
 
+COPY --from=quay.io/goswagger/swagger:v0.28.0 /usr/bin/swagger /usr/bin/goswagger
+COPY --from=quay.io/edge-infrastructure/swagger-codegen-cli:2.4.18 /opt/swagger-codegen-cli /opt/swagger-codegen-cli
```
Comment (adriengentil) [inline Dockerfile.assisted-service-build:4]: Why do we have our own version of `swagger-codegen-cli` ?

**Comment by osherdp** [comment]
> Is this new job in Prow will call `make ci-lint` like in Jenkins? In that case, we may want to remove the call to skipper in the target.

I thought about calling only ``generate-all`` to make it specific to code-generation
Like here: https://github.com/openshift/release/pull/27000/files#diff-36f1e2a28d2accb66765f2652cd6f9c01edfe4e2c5e930abdf787e01782c2d29R134
(and yes, all those git commands in there are a bit ugly, still unsure if we have a simpler way to check for differences)

File: Dockerfile.assisted-service-build (hunk 2 above)
Comment (osherdp) [reply Dockerfile.assisted-service-build:4]: we mainly needed something from quay.io to have less rate-limiters problems than on docker-hub

File: Dockerfile.assisted-service-build (hunk 1 above)
Comment (omertuc) [reply Dockerfile.assisted-service-build:16]: Oh sorry didn't realize this emoji is "confused" I thought it was "sad", I was just trying to show solidarity

Comment (osherdp) [reply Dockerfile.assisted-service-build:16]: your solidarity has been accepted and appreciated :smile:

**Pull request merged**